lib/crypto.js

```diff
@@ -652,6 +672,23 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// timingSafeEqual reuses the timing-safe Hmac.equals function (see above) for
+// comparison of non-hash inputs, like capability URLs or authentication tokens.
+// Note that the strength of the double-HMAC construction does not depend
+// on the unpredictability of the key. Since randomBytes can potentially
+// throw an exception, we use Math.random, which sacrifices nothing in timing
+// safety and benefits from never throwing an exception.
+// See discussion at
+// https://github.com/nodejs/node-v0.x-archive/issues/8560#issuecomment-60145816
+exports.timingSafeEqual = function(a, b) {
+  var key = new Buffer(32);
+  for (var i = 0; i < 32; i += 4) {
+    key.writeFloatBE(Math.random(), i);
```

Comment:
> Under what conditions?

Really just under abnormal conditions, like running out of memory, but you could also cause an error from C++ by giving it a fake Buffer.

I might be wrong here, but I think the documentation for `randomBytes` is out of date. Under normal circumstances, the only error thrown from `randomBytes` is from [`RAND_bytes()`](https://github.com/openssl/openssl/blob/9b86974e0c705ea321ddbc9a9d8562c894809e5b/doc/crypto/RAND_bytes.pod), which throws when it doesn't have enough entropy. However, with [`CheckEntropy`](https://github.com/nodejs/node/blob/212c9c086c526a0fdf8ac0bb0db36626f0534a0a/src/node_crypto.cc#L228-L259) (see https://github.com/nodejs/node/commit/f68a116c3c061151ab120f10fdc9230192d6b157), it should now block until the system has enough entropy for the OpenSSL pool (see log in https://github.com/nodejs/node/commit/e5e598060eb43faf2142184d523a04f0ca2d95c3). I think the documentation just needs to be updated - now that you bring it up, the docs are very confusing on that.